web/elm/src/Build/Header/Header.elm

```diff
@@ -63,76 +68,128 @@ header session model =
         , Views.Duration (duration session model)
         ]
     , rightWidgets =
-        if archived then
-            []
-
-        else
-            [ Views.Button
-                (if Concourse.BuildStatus.isRunning model.status then
-                    Just
-                        { type_ = Views.Abort
-                        , isClickable = True
-                        , backgroundShade =
-                            if
-                                HoverState.isHovered
-                                    AbortBuildButton
-                                    session.hovered
-                            then
-                                Views.Dark
-
-                            else
-                                Views.Light
-                        , backgroundColor = Concourse.BuildStatus.BuildStatusFailed
-                        }
-
-                 else if model.job /= Nothing then
+        Views.Button
+            (Just
+                { type_ = Views.ToggleComment
```

Comment:
If the user isn't authorized to set/read the build comment, we should hide this button